pkgs/by-name/au/audiveris/package.nix

```diff
@@ -0,0 +1,132 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  gradle_8,
+  makeWrapper,
+  graphicsmagick,
+  jre,
+  tesseract,
+  freetype,
+  testers,
+  audiveris,
+}:
+let
+  # Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
+  gradle = gradle_8;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "audiveris";
+  version = "5.5.3";
+
+  src = fetchFromGitHub {
+    owner = "Audiveris";
+    repo = "audiveris";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-GRKyQ+DqappqkbKuq5U+4ls96vdpPGPUGRHELjzfyo8=";
+  };
+
+  nativeBuildInputs = [
+    gradle
+    makeWrapper
+    graphicsmagick
+  ];
+
+  mitmCache = gradle.fetchDeps {
+    inherit (finalAttrs) pname;
+    data = ./deps.json;
+  };
+
+  __darwinAllowLocalNetworking = true; # required for mitm-cache on Darwin
+
+  gradleFlags = [
+    "-Dfile.encoding=utf-8"
+
+    # The Audiveris build assumes a git checkout.
+    # The flags below will make it work with the tarball, too.
+    "-PprogramBuild=${finalAttrs.version}"
+    "-x=getCommit"
+    "-x=getCommitShort"
+  ];
+
+  doCheck = true;
+
+  dontStrip = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    echo 'gradle.projectsLoaded {
+      rootProject.allprojects {
+        afterEvaluate {
+          if (it.path == ":app") {
+            tasks.register("writeClasspathFile") {
+              file("classpath.txt").text =
+                configurations.runtimeClasspath.files.collect { it.name }.join("\n")
+            }
+          }
+        }
+      }
+    }' >writeClasspathFile.gradle
+    gradle -q --init-script writeClasspathFile.gradle writeClasspathFile
+
+    mkdir -p -- "$out"
+    tar xC "$out" \
+      -f 'app/build/distributions/app-${finalAttrs.version}.tar' \
+      --transform 's|^app-${finalAttrs.version}/lib|share/java|' \
+      'app-${finalAttrs.version}/lib/'
+
+    mapfile -t jars <classpath.txt
+    cp="$out/share/java/audiveris.jar"
+    for jar in "''\${jars[@]}"; do
+      cp+=":$out/share/java/$jar"
+    done
+
+    makeWrapper ${lib.getExe jre} $out/bin/audiveris \
+      --set-default TESSDATA_PREFIX '${tesseract}/share/tessdata' \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ freetype ]}" \
+      --add-flags "-cp '$cp' Audiveris"
+
+    install -D -m0644 app/res/icon-64.png "$out/share/icons/hicolor/64x64/apps/audiveris.png"
+    for size in 16 24 32 48; do
+      # set modify/create for reproducible builds
+      gm convert app/res/icon-64.png -scale $size +set date:create +set date:modify icon.png
+      install -D -m0644 icon.png "$out/share/icons/hicolor/''${size}x$size/apps/audiveris.png"
+    done
```

Comment:
I am not sure you need the smaller sizes

Reply:
Not sure either what the typical sizes should be. Will do whatever you think is appropriate.

Another option would be to use the upstream ico file and take all the sizes from there, e.g. like this:

```bash
mapfile -t iconSizes < <(magick identify -format '%wx%h\n' app/res/all-icons-256.ico)
i=0
for iconSize in "''${iconSizes[@]}"; do
  magick convert app/res/all-icons-256.ico"[$((i++))]" +set date:create +set date:modify icon.png
  install -D -m0644 icon.png "$out/share/icons/hicolor/$iconSize/apps/audiveris.png"
done
```

Reply:
Extracting them from the ico sounds good, yes. FWIW I don't know of any software that would need X px icons if Y px icons (Y > X) are available. I see lots of derivations that just install one size (256px for example), and nobody complained so far.